### ISSUE TITLE
Mozilla intends to deprecate OpenSearch

### ISSUE THREAD

**Issue by sojusnik** [ssborbis/ContextSearch-web-ext#219]
See [here](https://blog.mozilla.org/addons/2019/10/15/search-engine-add-ons-to-be-removed-from-addons-mozilla-org/), [here](https://discourse.mozilla.org/t/deprecation-of-opensearch-xml-add-ons-current-users/46868) and [here](https://www.soeren-hentzschel.at/firefox/firefox-opensearch-unterstutzung/).

Do you plan to support the creation and administration of search engines as WebExtensions?

**Comment by ssborbis** [comment]
I'll have to look at the API and see what's possible. There's very little info right now. Judging from the comments, people are not happy about it.

**Comment by KrasnayaPloshchad** [comment]
So it would be exciting if this add-on continue to support OpenSearch plugins.

**Comment by ssborbis** [comment]
@KrasnayaPloshchad 

Mozilla has already depreciated window.external.AddSearchProvider() but I've had a workaround in place for the past several versions. Are you having problems installing OpenSearch engines?

**Comment by ssborbis** [comment]
@KrasnayaPloshchad 

Sorry, I think I'm conflating engines with add-ons. If I'm remembering correctly, Firefox had an odd way of handling search engines AS add-ons, and I don't have any experience with that method. Regardless, if you're having problems let me know.

**Comment by KrasnayaPloshchad** [comment]
>Are you having problems installing OpenSearch engines?

No, I assume if Mozilla eventually decided to drop this feature in the future you can make your add-on continually support OpenSearch to honour users’ wishes.
